src/reducers/tabs.js

```diff
@@ -75,22 +80,32 @@ export function removeSourcesFromTabList(tabs: TabList, sources: Source[]) {
  */
 function updateTabList(
   tabs: TabList,
-  { url, framework = null, isOriginal = false }
+  { url, framework = null, sourceId, isOriginal = false }
 ) {
-  const currentIndex = tabs.findIndex(tab =>
-    isSimilarTab(tab, url, isOriginal)
-  );
+  // Set currentIndex to -1 for URL-less tabs so that they aren't
+  // filtered by isSimilarTab
+  const currentIndex = url
+    ? tabs.findIndex(tab => isSimilarTab(tab, url, isOriginal))
+    : -1;
 
```

Comment:
During manual testing I noticed that only one URL-less source tab would appear, and quickly figured out this was the problem.  This snippet was finding a tab without a URL, and saying, "OK, we already have this, don't create a new tab"